pkg/controller/llmisvc/DEV.md

```diff
@@ -715,4 +715,204 @@ curl -sS -X POST http://localhost:8001/$NS/$LLM_ISVC_NAME/v1/completions   \
+  name: rhcl-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+---
+kind: OperatorGroup
+apiVersion: operators.coreos.com/v1
+metadata:
+  name: kuadrant
+  namespace: kuadrant-system
+spec:
+  upgradeStrategy: Default
+EOF
+
+while [ $(kubectl get pod -n kuadrant-system  | wc -l) -le 1 ]; 
+do
+  echo "⏳ waiting for Kuadrant Pod to appear…"
+  sleep 10
+done
+kubectl wait --for=condition=ready pod -l app=kuadrant -n kuadrant-system --timeout 150s
+
+kubectl apply -f - <<EOF
+apiVersion: kuadrant.io/v1beta1
+kind: Kuadrant
+metadata:
+  name: kuadrant
+  namespace: kuadrant-system
+EOF
+```
+
+**Create Pull Secret**
+
+Refer to [this doc](https://docs.redhat.com/en/documentation/red_hat_connectivity_link/1.0/html-single/installing_connectivity_link_on_openshift/index#auth-registry-wasm-plugin)
+
+You can get the secret from [this site](https://access.redhat.com/terms-based-registry/) and create it in `api-gateway` namespace and gateway namespace(`openshift-ingress`)
+
+```shell
+kubectl create ns api-gateway
+
+kubectl create secret docker-registry wasm-plugin-pull-secret -n api-gateway \ --docker-server=registry.redhat.io \ --docker-username=<your-registry-service-account-username> \ --docker-password=<your-registry-service-account-password>
+
+kubectl create secret docker-registry wasm-plugin-pull-secret -n openshift-ingress \ --docker-server=registry.redhat.io \ --docker-username=<your-registry-service-account-username >\ --docker-password=<your-registry-service-account-password>
+
+```
+
+**Update gateway label**
+```shell
+kubectl label gateway openshift-ai-inference -n openshift-ingress kuadrant.io/gateway="true"
+```
+
+**Authentication with KubernetesTokenReview**
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: kuadrant.io/v1
+kind: AuthPolicy
+metadata:
+  name: llm-test-authn
+  namespace: $TEST_NS
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: $LLM_ISVC_NAME-kserve-route
+  defaults:
+    strategy: merge
+    rules:
+      authentication:
+        kubernetes-user:
+          credentials:
+            authorizationHeader: {}
+          kubernetesTokenReview:
+            audiences:
+            - https://kubernetes.default.svc
+          metrics: false
+          priority: 0
+      authorization:
+        kubernetes-rbac:
+          kubernetesSubjectAccessReview:
+            resourceAttributes:
+              group:
+                value: gateway.networking.k8s.io/v1
+              name:
+                value: $LLM_ISVC_NAME-kserve-route
+              namespace:
+                value: $TEST_NS
+              resource:
+                value: httproutes
+              subresource:
+                value: ""
+              verb:
+                value: get
+            user:
+              selector: auth.identity.user.username
+          metrics: false
+          priority: 0
+EOF
+
+```
+
+**Validation**
+
+- Using external hostname (ROSA or metalLB)
```

Comment:
Do we need to mention ROSA or specifics setup since this works with more real OCP clusters?

Reply:
Unfortunately, it is needed because ROSA is kind of special. It already has external LB like metalLB so it can use llmisvc status external url.